File: rb/features.rb
# DwNews SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module DwNewsFeatures
  def self.make_feature(name)
    case name
    when "base"
      DwNewsBaseFeature.new
    when "ratelimit"
      DwNewsRatelimitFeature.new
    when "retry"
      DwNewsRetryFeature.new
    when "test"
      DwNewsTestFeature.new
    when "timeout"
      DwNewsTimeoutFeature.new
    else
      DwNewsBaseFeature.new
    end
  end
end

File: py/features.py
# DwNews SDK feature factory

from feature.base_feature import DwNewsBaseFeature
from feature.test_feature import DwNewsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DwNewsBaseFeature(),
        "test": lambda: DwNewsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
